core/src/main/java/com/testaem/aem/core/models/impl/SinglePageNewsImpl.java

```diff
@@ -50,4 +110,24 @@ public String getImagePath() {
     public String getDate() {
         return this.date == null ? simpleDateFormat.format(new Date()) : simpleDateFormat.format(date);
     }
+
+    public String getPagePath() {
+
+        List<String> selectors = Arrays.asList(request.getRequestPathInfo().getSelectors());
+
+        boolean fullPage = selectors.contains("fullPage");
+
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
+        Page containingPage = pageManager.getContainingPage(request.getResource());
+        ;
+        Page landingPage = containingPage.getParent(2);
```

Comment:
Magic numbers, BTW)
Magic numbers in code are bad. Consider creating a constant saying like `STEPS_TO_LANDING_PAGE = 2;`